src/pages/Signup/Signup.js

```diff
@@ -1,8 +1,293 @@
-import React from 'react';
+import React, { useState } from 'react';
+import { Link } from 'react-router-dom';
+import { AlertCircle, ArrowLeft, CheckCircle } from 'react-feather';
 import './Signup.scss';
 
-function Signup() {
-  return <div>Signup.js</div>;
-}
+const Signup = () => {
+  const [inputs, setInputs] = useState({
+    name: '',
+    birth: '',
+    phone: '',
+    gender: '',
+    email: '',
+    address: '',
+    password: '',
+  });
+
+  const [checkboxes, setCheckboxes] = useState({
+    terms: false,
+    privacy: false,
+  });
+
+  const { terms, privacy } = checkboxes;
+
+  const { name, birth, phone, gender, email, address, password } = inputs;
+
+  const allFieldFilled =
+    name && birth && phone && email && address && password && terms && privacy;
+
+  const handleInput = event => {
+    const { name, value } = event.target;
+    setInputs({
+      ...inputs,
+      [name]: value,
+    });
+  };
+
+  const validations = {
+    name: name.length >= 3,
+    birth: birth.match(/^(\d{4})(\/|-)(\d{1,2})(\/|-)(\d{1,2})$/),
+    phone: phone.match(/^01([0|1|6|7|8|9])-?([0-9]{3,4})-?([0-9]{4})$/),
+    email: email.match(/^\S+@\S+\.\S+$/),
+    address: address.length >= 5,
+    password: password.match(/^(?=.*[A-Za-z])(?=.*[0-9])(?=.{8,20})/),
+  };
+
+  const messages = {
+    name: ['사용 가능한 이름입니다.', '이름을 입력해주세요.'],
+    birth: [
+      '사용 가능한 생일입니다.',
+      '생년월일을 YYYY-MM-DD 형식으로 입력해주세요.',
+    ],
+    phone: [
+      '사용 가능한 핸드폰 번호입니다.',
+      '올바른 핸드폰 번호를 입력해주세요.',
+    ],
+    email: ['사용 가능한 이메일입니다.', '올바른 이메일 정보를 입력해주세요.'],
+    address: ['사용 가능한 주소입니다.', '올바른 주소 정보를 입력해주세요.'],
+    password: [
+      '사용 가능한 비밀번호입니다.',
+      '8자 이상의 영문+숫자 조합 비밀번호를 입력해주세요.',
+    ],
+  };
+
+  const allValueValidated =
+    allFieldFilled &&
+    validations.name &&
+    validations.birth &&
+    validations.phone &&
+    validations.phone &&
+    validations.email &&
+    validations.password;
```

Comment:
```suggestion
  const isAllValid =
    Object.values(validations).every(valid => valid) &&
    Object.values(checkboxes).every(valid => valid);
```

Object.values 는 객체에서 값을 배열로 변경해주는 메서드고
every는 배열에 요소에서 조건에 모두 부합하면 true 하나라도 부합하지 않으면 false를 내보내는 메서드입니다.

이 두 메서드를 이용해서 모든 조건에 통과했는지 확인할 수 있겠습니다!